File: URI/ED/1110.cpp
#include <bits/stdc++.h>
#define FOR(i, j, k) for(int i = j; i <k ; i++)
#define pb push_back


using namespace std;

int main(){
	int n;
	while(true){
		cin >> n;
		if(!n)break;
		vector<int> si;
		for(int i = n; i >= 1; i--){
			si.pb(i);
		}
		cout << "Discarded cards: " << si[si.size() - 1];
		si.pop_back();
		vector<int> temp;
		temp.pb(si[si.size()-1]);
		
		while(si.size() > 1){
			
			
			cout << ", " << si[si.size() -1 ] ;
			si.pop_back();
			
			for (int i = 0; i < si.size(); i++){
				temp.pb(si[i]);
			}
			si = temp;
		}
		cout << "Remaining card: " << si[0] << endl;
	}
	return 0;
}

File: URI/ED/1110.py
while True:
	n = int(input())
	if n == 0:
		break
	pilha = []
	for i in range(n, 0, -1):
		pilha.append(i)
	print("Discarded cards: " + str(pilha[-1]), end='')
	pilha.pop()
	pilha = [pilha[-1]] + pilha
	pilha.pop()
	while len(pilha) > 1:
		print(", " + str(pilha[-1]), end='')
		pilha.pop()
		pilha = [pilha[-1]] + pilha
		pilha.pop()
	print("\nRemaining card: " + str(pilha[0]))
